### PR TITLE
AA [WIP]

### PR DESCRIPTION
**Pull request by DonGambas**
- export `EOAWalletManager`
- add export`SmartAccountManager`
- create SmartAccountManager instances for kernel, safe and lightaccount wallets.

### CODE REVIEW

**Review by rweald — COMMENTED** [review]
I think we probably want to export a shared interface for the core overlapping methods between EOAWalletManager, and the smart account stuff. For the common methods

File: src/smart_account.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { Contract } from 'ethers';
+
+export interface SmartAccountManager {
+  getAddress: (address?: string, salt?: number) => Promise<string>;
```
Comment (rweald) [inline src/smart_account.ts:4]: What would passing in the optional `address` be used for?

Comment (DonGambas) [reply src/smart_account.ts:4]: the smart contract address is generated using an address + salt, so in our case we'd default to using the current user eoa [eoa at slot 0 on mnemonic 0], but if we start to support many EOAs and mnemonics might be nice to allow for smart contract wallet generation derived from another EOA

**Comment by rweald** [comment]
We've changed product requirements & devex since this was opened. Going to close in favor of our new approach.

**Pull request closed without merging**